src/mono/wasm/runtime/library_mono.js

```diff
@@ -536,109 +535,323 @@ var MonoSupportLib = {
+					if (sourcePrefix === "./")
+						sourcePrefix = "";
+
+					var attemptUrl;
+					if (sourcePrefix.trim() === "") {
+						// FIXME: Disgusting magic to match old search behavior
+						if (asset.behavior === "assembly")
+							attemptUrl = locateFile (args.deploy_prefix + "/" + asset.name);
+						else
+							attemptUrl = asset.name;
+					} else {
+						attemptUrl = sourcePrefix + asset.name;
 					}
-				}).then (function (blob) {
-					var asm = new Uint8Array (blob);
-					var memory = Module._malloc(asm.length);
-					var heapBytes = new Uint8Array(Module.HEAPU8.buffer, memory, asm.length);
-					heapBytes.set (asm);
-					mono_wasm_add_assembly (file_name, memory, asm.length);
-
-					//console.log ("MONO_WASM: Loaded: " + file_name);
-					--pending;
-					if (pending == 0) {
-						MONO.loaded_files = loaded_files;
-						var load_runtime = Module.cwrap ('mono_wasm_load_runtime', null, ['string', 'number']);
-
-						console.log ("MONO_WASM: Initializing mono runtime");
-						if (ENVIRONMENT_IS_SHELL || ENVIRONMENT_IS_NODE) {
-							try {
-								load_runtime (vfs_prefix, enable_debugging);
-							} catch (ex) {
-								print ("MONO_WASM: load_runtime () failed: " + ex);
-								var err = new Error();
-								print ("MONO_WASM: Stacktrace: \n");
-								print (err.stack);
-
-								var wasm_exit = Module.cwrap ('mono_wasm_exit', null, ['number']);
-								wasm_exit (1);
-							}
+
+					try {
+						if (asset.name === attemptUrl) {
+							console.log ("Attempting to fetch", attemptUrl);
 						} else {
-							load_runtime (vfs_prefix, enable_debugging);
+							console.log ("Attempting to fetch", attemptUrl, "for", asset.name);
 						}
-						MONO.mono_wasm_runtime_ready ();
-						loaded_cb ();
+						var fetch_promise = fetch_file_cb (attemptUrl);
+						fetch_promise.then (handleFetchResponse);
+					} catch (exc) {
+						console.error ("MONO_WASM: Error fetching " + attemptUrl, exc);
+						attemptNextSource ();
 					}
-				});
+				};
+
+				attemptNextSource ();
 			});
 		},
 
+		_globalization_init: function (args, ctx) {
+			var invariantMode = false;
+
+			if (args.globalization_mode === "invariant")
+				invariantMode = true;
+
+			if (!invariantMode) {
+				if (ctx.num_icu_assets_loaded_successfully > 0) {
+					console.log ("MONO_WASM: ICU data archive(s) loaded, disabling invariant mode");
+				} else if (args.globalization_mode !== "icu") {
+					console.log ("MONO_WASM: ICU data archive(s) not loaded, using invariant globalization mode");
+					invariantMode = true;
+				} else {
+					console.error ("MONO_WASM: ERROR: invariant globalization mode is inactive and no ICU data archives were loaded.");
+				}
+			}
+
+			this.mono_wasm_setenv("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", invariantMode ? "1" : "0");
+		},
+
 		mono_wasm_get_loaded_files: function() {
-			console.log(">>>mono_wasm_get_loaded_files");
-			return this.loaded_files;
+			throw new Error("not implemented");
```

Comment:
This is consumed by the debugger and can't simply be removed.  There are some other changes in mono/mono that will need to be merged in.

Reply:
OK, I think get_loaded_files should work. What changes need to be merged in? How do I do it? I thought we had repo mirroring in place.

Reply:
some parts are not mirrored, the updated code will work for this pr